imageio/imageio-webp/src/main/java/com/twelvemonkeys/imageio/plugins/webp/WebPImageReader.java

```diff
@@ -527,7 +527,8 @@ private void readAlpha(BufferedImage destination, ImageReadParam param, final in
             System.out.println("compression: " + compression);
         }
 
-        WritableRaster alphaRaster = destination.getAlphaRaster();
+        // Alpha raster must have same dimensions as the source, because of gradient filtering.
+        WritableRaster alphaRaster = Raster.createInterleavedRaster(DataBuffer.TYPE_BYTE, width, height, 1, null);
```

Comment:
I think we should be able to get away with *not* creating another full-size temporary raster here. Instead we can do the filtering on the `tempRaster` below, by creating a *writable* child raster on line 544, and do the alpha filtering on that.

Something like this:

```java
        WritableRaster alphaRaster = destination.getAlphaRaster();

        switch (compression) {
            case 0:
                readUncompressedAlpha(alphaRaster);
                break;

            case 1:
                // Simulate header
                imageInput.seek(imageInput.getStreamPosition() - 5);

                WritableRaster tempRaster = Raster.createInterleavedRaster(DataBuffer.TYPE_BYTE, width, height, 4, null);
                readVP8Lossless(tempRaster, null, width, height);

                // Copy from green (band 1) in temp to alpha in destination
                WritableRaster alphaChannel = tempRaster.createWritableChild(0, 0, tempRaster.getWidth(), tempRaster.getHeight(), 0, 0, new int[]{1});
                alphaFilter(alphaChannel, filtering);
                alphaRaster.setRect(alphaChannel); // TODO: Replace this with copyIntoRasterWithParams(alphaChannel, alphaRaster, param);
                break;

            default:
                processWarningOccurred("Unknown WebP alpha compression: " + compression);
                opaqueAlpha(alphaRaster);
                break;
        }

```

...and extracting the current alpha filtering to a separate method:

```java
    private void alphaFilter(WritableRaster alphaRaster, int filtering) {
        if (filtering != AlphaFiltering.NONE) {
            for (int y = 0; y < alphaRaster.getHeight(); y++) {
                for (int x = 0; x < alphaRaster.getWidth(); x++) {
                    int predictorAlpha = getPredictorAlpha(alphaRaster, filtering, y, x);
                    alphaRaster.setSample(x, y, 0, alphaRaster.getSample(x, y, 0) + predictorAlpha % 256);
                }
            }
        }
    }

```

I'm not sure if alpha filtering is supported for uncompressed alpha, and we don't support it anyway (if you have a sample file with uncompressed alpha, please send it to me! 😀 ).

Reply:
> > I'm not sure if alpha filtering is supported for uncompressed alpha, and we don't support it anyway (if you have a sample file with uncompressed alpha, please send it to me! 😀 ).
> 
> I checked all WebP files I have here, and couldn't find any with uncompressed alpha. I will try to collect new ones, and let you know if I can find any of these.

Looking at a larger set of WebPs (~40K files), I found 3 that are reported as having "alpha compression = 0".
I am not sure if they actually have uncompressed alpha, or are false detections caused by some issue in the files.
They do enter in `readUncompressedAlpha()` while decoding.
[possibly-uncompressed-alpha-samples.zip](https://github.com/haraldk/TwelveMonkeys/files/10985381/possibly-uncompressed-alpha-samples.zip)